crates/libs/core/src/sync/cancel.rs

```diff
@@ -466,13 +479,14 @@ mod test {
         async fn get_data_delay(
             &self,
             delay: Duration,
+            ignore_cancel: bool,
             token: Option<CancellationToken>,
         ) -> crate::Result<String> {
             if delay.is_zero() {
                 return Ok(self.get_data());
             }
-            match token {
-                Some(t) => {
+            match (token, ignore_cancel) {
+                (Some(t), false) => {
                     select! {
                         _ = t.cancelled() => {
```

Comment:
@rsajwani For example there in the implementation, the cancel signal is received and we cancel the operation immediately without processing the request with an error code.
If we ignore this cancel signal, as we do in some other test branch, the delay would happen irrespective to the signal.